### DISCUSSION TITLE
How obtain the id after creation

### DISCUSSION THREAD

**Discussion by eaziweb95** [filamentphp/filament#3045] [Help] [answered]
I'd like ton send a email after creation, i used the afterCreate method as below
but how can i obtain the id of the record just created ? is it data['id'] ?

    protected function afterCreate(array $data): array
    {
        $demande = Demande::findOrFail($data['id']);

        try {
            Mail::send(new DemandeDepose($demande));
        }
        catch (Exception $exception) {
            Log::channel('emailError')->error($exception->getMessage(),  ['demande' => $demande]);
        }
    }

**Comment by danharrin** [comment, marked as answer]
`$this->record`

**Reply by eaziweb95** [reply]
Thanks a lot
